File: MultiLanguageSandbox/src/test/deps/pascal/46.pas
{$ASSERTIONS ON}

uses
  SysUtils;
(*
  Given an integer n, this function calculates the sum of all numbers from 1 to n that
  are multiples of either 3 or 5. Each qualifying number is counted only once, even if
  it is a multiple of both 3 and 5.

  Example usage:
    >>> sumOfMultiples(10)
    Returns 33, which represents the sum of 3, 5, 6, 9, and 10.
*)

function sumOfMultiples(n: integer): integer;
var
  i, sum: integer;
begin
  sum := 0;
  for i := 1 to n do
  begin
    if (i mod 3 = 0) or (i mod 5 = 0) then
      sum := sum + i;
  end;
  sumOfMultiples := sum;
end;
procedure testSumOfMultiples;
begin
    Assert(sumOfMultiples(10) = 33);
    Assert(sumOfMultiples(15) = 60);
    Assert(sumOfMultiples(20) = 98);
    Assert(sumOfMultiples(5) = 8);
    Assert(sumOfMultiples(3) = 3);
    Assert(sumOfMultiples(6) = 14);
    Assert(sumOfMultiples(9) = 23);
    Assert(sumOfMultiples(12) = 45);
    Assert(sumOfMultiples(17) = 60);
    Assert(sumOfMultiples(21) = 119);
    Assert(sumOfMultiples(25) = 168);
end;

begin
    testSumOfMultiples;
end.

File: MultiLanguageSandbox/src/test/deps/c/46.c

#include <assert.h>
#include <stdio.h>
/*
Given an integer n, write a function to compute the sum of all numbers from 1 to n that are multiples of either 3 or 5. If a number is a multiple of both 3 and 5, it should only be counted once.
For example:
    >>> sumOfMultiples(10)
    33 // (3 + 5 + 6 + 9 + 10)
*/
int sumOfMultiples(int n) {
    int sum = 0;
    for (int i = 1; i <= n; i++) {
        if (i % 3 == 0 || i % 5 == 0) {
            sum += i;
        }
    }
    return sum;
}
int main()
{
    assert(sumOfMultiples(10) == 33);
    assert(sumOfMultiples(15) == 60);
    assert(sumOfMultiples(20) == 98);
    assert(sumOfMultiples(5) == 8);
    assert(sumOfMultiples(3) == 3);
    assert(sumOfMultiples(6) == 14);
    assert(sumOfMultiples(9) == 23);
    assert(sumOfMultiples(12) == 45);
    assert(sumOfMultiples(17) == 60);
    assert(sumOfMultiples(21) == 119);
    assert(sumOfMultiples(25) == 168);
    return 0;
}